physics/transforms_body.hpp

```diff
@@ -217,22 +223,22 @@ Transforms<FromFrame, ThroughFrame, ToFrame>::BarycentricRotating(
   };
 
   transforms->second_ =
-      [to_primary_trajectory, to_secondary_trajectory](
+      [&primary, &secondary, to_trajectory](
           Instant const& t,
           DegreesOfFreedom<ThroughFrame> const& through_degrees_of_freedom,
           Trajectory<ThroughFrame> const* trajectory) ->
       DegreesOfFreedom<ToFrame> {
     DegreesOfFreedom<ToFrame> const& last_primary_degrees_of_freedom =
-        to_primary_trajectory().last().degrees_of_freedom();
+        (primary.*to_trajectory)().last().degrees_of_freedom();
```

Comment:
put the result of `LazyTrajectory` in a local variable, if it changes between calls here some threading magic is happening that violates the contract anyway.